modules/decrypt-node/src/verify_stream.ts

```diff
@@ -251,28 +278,36 @@ export class VerifyStream extends PortableTransformWithType {
   }
 
   _flush(callback: Function) {
-    const { finalAuthTagReceived } = this._verifyState
-    /* Precondition: All ciphertext MUST have been received.
-     * The verify stream has ended,
-     * there will be no more data.
-     * Therefore we MUST have reached the end.
-     */
-    if (!finalAuthTagReceived) return callback(new Error('Incomplete message'))
-    /* Check for early return (Postcondition): If there is no verify stream do not attempt to verify. */
-    if (!this._verify) return callback()
     try {
-      const { signatureInfo } = this._verifyState
-      /* Precondition: The signature must be well formed. */
-      const { buffer, byteOffset, byteLength } = deserializeSignature(
-        signatureInfo
-      )
-      const signature = Buffer.from(buffer, byteOffset, byteLength)
-      const isVerified = this._verify.awsCryptoVerify(signature)
-      /* Postcondition: The signature must be valid. */
-      needs(isVerified, 'Invalid Signature')
-      callback()
-    } catch (e) {
-      callback(e)
+      const { finalAuthTag, signature } = this._verifyState
+      /* Precondition: All ciphertext MUST have been received.
+       * The verify stream has ended,
+       * there will be no more data.
+       * Therefore we MUST have reached the end.
+       */
+      needs(finalAuthTag, 'Incomplete message')
+
+      /* Verifying the signature here
+       * **before** the authTag is released
+       * make the VerifyStream more composable.
+       * If I need to verify something
+       * without decrypting, this is possible.
+       */
+      if (this._verify) {
+        /* Precondition: A complete signature is required to verify. */
+        needs(signature, 'Incomplete signature')
+        const { buffer, byteOffset, byteLength } = signature
+        const isVerified = this._verify.awsCryptoVerify(
+          Buffer.from(buffer, byteOffset, byteLength)
+        )
+        /* Postcondition: The signature must be valid. */
+        needs(isVerified, 'Invalid Signature')
```

Comment:
nit: s/S/s/

Reply:
```suggestion
        needs(isVerified, 'Invalid signature')
```

Reply:
Yes, and the corresponding tests should change as well.